jackson-databind/src/main/java/io/micronaut/jackson/modules/BeanIntrospectionModule.java

```diff
@@ -310,50 +332,18 @@ public BeanSerializerBuilder updateBuilder(SerializationConfig config, BeanDescr
                     }
 
                     final List<BeanPropertyWriter> newProperties = new ArrayList<>(properties);
-                    Map<String, BeanProperty> named = new LinkedHashMap<>(properties.size());
+                    Map<String, BeanProperty<Object, Object>> named = new LinkedHashMap<>();
                     for (BeanProperty<Object, Object> beanProperty : beanProperties) {
-                        final Optional<String> n = beanProperty.stringValue(JsonProperty.class);
-                        n.ifPresent(s -> named.put(s, beanProperty));
+                        named.put(getName(config, namingStrategy, beanProperty), beanProperty);
                     }
                     for (int i = 0; i < properties.size(); i++) {
                         final BeanPropertyWriter existing = properties.get(i);
 
-                        final Optional<BeanProperty<Object, Object>> property;
-                        final String existingName = existing.getName();
-                        if (named.containsKey(existingName)) {
-                            property = Optional.of(named.get(existingName));
-                        } else {
-                            property = introspection.getProperty(existingName);
-                        }
+                        final Optional<BeanProperty<Object, Object>> property = Optional.ofNullable(named.get(existing.getName()));
                         // ignore properties that are @JsonIgnore, so that we don't replace other properties of the
                         // same name
                         if (property.isPresent() && !property.get().isAnnotationPresent(JsonIgnore.class)) {
                             final BeanProperty<Object, Object> beanProperty = property.get();
-                            if (isResource) {
```

Comment:
I assume we don't need this due to : 

https://github.com/micronaut-projects/micronaut-core/blob/3.4.x/http/src/main/java/io/micronaut/http/hateoas/Resource.java#L54

and: 

https://github.com/micronaut-projects/micronaut-core/blob/3.4.x/http/src/main/java/io/micronaut/http/hateoas/Resource.java#L46

Reply:
We have a test which verifies this: 

BeanIntrospectionModuleSpec::"void "test that introspected serialization of errors works"